File: c/exercism/difference_of_squares/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int square_of_sum(unsigned int number)
{
	unsigned int sum = 0;

	for (unsigned int i = 1; i <= number; i++)
	{
		sum += i;
	}

	return sum * sum;
}

unsigned int sum_of_squares(unsigned int number)
{
	unsigned int sum = 0;

	for (unsigned int i = 1; i <= number; i++)
	{
		sum += i * i;
	}

	return sum;
}

unsigned int difference_of_squares(unsigned int number)
{
	unsigned int squaredSum = square_of_sum(number);
	unsigned int sumOfSquareds = sum_of_squares(number);

	return squaredSum - sumOfSquareds;
}

File: go/exercism/difference_of_squares/difference_of_squares.go
package main

func SquareOfSum(number int) int {
	sum := 0

	for i := 1; i <= number; i++ {
		sum += i
	}

	return sum * sum
}

func SumOfSquares(number int) int {
	sum := 0

	for i := 1; i <= number; i++ {
		sum += i * i
	}

	return sum
}

func DifferenceOfSquares(number int) int {
	return SquareOfSum(number) - SumOfSquares(number)
}
